clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1126,28 +1139,43 @@ class ArchiveFileHandler final : public FileHandler {
         continue;
       }
 
+      std::unique_ptr<FileHandler> FHP{nullptr};
+      std::unique_ptr<MemoryBuffer> Buf{nullptr};
+      StringRef Ext("o");
+      if (BundlerConfig.FilesType == "aocr" ||
+          BundlerConfig.FilesType == "aocx")
+        Ext = BundlerConfig.FilesType;
+
       auto BinOrErr = C.getAsBinary();
       if (!BinOrErr) {
+        /* Not a recognized binary file.  Specifically not an object file */
         if (auto Err = isNotObjectErrorInvalidFileType(BinOrErr.takeError()))
           return Err;
-        continue;
-      }
-
-      auto &Bin = BinOrErr.get();
-      if (!Bin->isObject())
-        continue;
-
-      auto Obj = std::unique_ptr<ObjectFile>(cast<ObjectFile>(Bin.release()));
-      auto Buf = MemoryBuffer::getMemBuffer(Obj->getMemoryBufferRef(), false);
 
-      auto ChildNameOrErr = C.getName();
-      if (!ChildNameOrErr)
-        return ChildNameOrErr.takeError();
+        if (BundlerConfig.FilesType == "aoo") {
+          /* Handle BC Files */
+          Ext = "bc";
+          FHP = std::make_unique<BinaryFileHandler>(BundlerConfig);
+          auto MR = C.getMemoryBufferRef();
+          if (!MR) {
+            dbgs() << "No memory buffer\n";
```

Comment:
Will this need to be removed before merge?

Reply:
Changed to assert.